source/OrderedTypes/PredicativeLFP.lagda

```diff
@@ -0,0 +1,1711 @@
+   ＝ˢ-equiv-＝ {x} {y} = resizing-condition (l-small x y)
+
+   ＝ˢ-to-＝ : {x y : ⟨ L ⟩} → (x ＝ˢ y) → (x ＝ y)
+   ＝ˢ-to-＝ = ⌜ ＝ˢ-equiv-＝ ⌝
+ 
+   ＝-to-＝ˢ : {x y : ⟨ L ⟩} → (x ＝ y) → (x ＝ˢ y)
+   ＝-to-＝ˢ = ⌜ ＝ˢ-equiv-＝ ⌝⁻¹
+
+   ＝ˢ-refl : {x : ⟨ L ⟩} → x ＝ˢ x
+   ＝ˢ-refl = ＝-to-＝ˢ refl
+
+  dense-implies-bounded : (f : ⟨ L ⟩ → ⟨ L ⟩)
+                        → is-monotone-endomap L f
+                        → is-dense f
+                        → Σ ϕ ꞉ 𝓟 {𝓤 ⊔ 𝓥} (B × ⟨ L ⟩) ,
+                          Σ bnd ꞉ is-bounded ϕ ,
+                           ((a : ⟨ L ⟩)
+                            → (Γ ϕ (bounded-implies-local ϕ bnd)) a ＝ f a)
+  dense-implies-bounded f f-mono (I , γ , f-dense) = (ϕ , bnd , H)
+   where
+    ϕ : 𝓟 {𝓤 ⊔ 𝓥} (B × ⟨ L ⟩)
+    ϕ (b , a') =
+     (Lift 𝓤 ((Ǝ i ꞉ I , b ≤ᴮ f (γ i) × γ i ＝ˢ a') holds)
+      , equiv-to-prop (Lift-≃ 𝓤 ((Ǝ i ꞉ I , b ≤ᴮ f (γ i) × γ i ＝ˢ a') holds))
+                      (holds-is-prop (Ǝ i ꞉ I , b ≤ᴮ f (γ i) × γ i ＝ˢ a')))
+    bnd : is-bounded ϕ
+    bnd = (ϕ-small , (I , (λ z → small-↓ᴮ (γ z)) , covering-cond))
+     where
+      ϕ-small : (a : ⟨ L ⟩) → (b : B) → (ϕ (b , a) holds) is 𝓥 small
+      ϕ-small a b = ((Ǝ i ꞉ I , b ≤ᴮ f (γ i) × γ i ＝ˢ a) holds
+                    , ≃-Lift 𝓤 ((Ǝ i ꞉ I , b ≤ᴮ f (γ i) × γ i ＝ˢ a) holds))
+      covering-cond : (a : ⟨ L ⟩)
+                    → (b : B)
+                    → (b , a) ∈ ϕ
+                    → (Ǝ i ꞉ I , small-↓ᴮ (γ i) ↠ ↓ᴮ L β a) holds
+      covering-cond a b = demote-equiv-to-surj ∘ transport-lemma ∘ unlift-ϕ
+       where
+        unlift-ϕ : (b , a) ∈ ϕ → (Ǝ i ꞉ I , b ≤ᴮ f (γ i) × γ i ＝ˢ a) holds
+        unlift-ϕ = ⌜ Lift-≃ 𝓤 ((Ǝ i ꞉ I , b ≤ᴮ f (γ i) × γ i ＝ˢ a) holds) ⌝
+        transport-lemma : (Ǝ i ꞉ I , b ≤ᴮ f (γ i) × γ i ＝ˢ a) holds
+                        → (Ǝ i ꞉ I , small-↓ᴮ (γ i) ≃ ↓ᴮ L β a) holds
+        transport-lemma =
+         ∥∥-rec (holds-is-prop (Ǝ i ꞉ I , small-↓ᴮ (γ i) ≃ ↓ᴮ L β a))
+                (λ (i , o , eq)
+                 → ∣ (i , transport (λ z → small-↓ᴮ (γ i) ≃ ↓ᴮ L β z)
+                                    (＝ˢ-to-＝ eq)
+                                    small-↓ᴮ-≃-↓ᴮ) ∣)
+        demote-equiv-to-surj : (Ǝ i ꞉ I , small-↓ᴮ (γ i) ≃ ↓ᴮ L β a) holds
+                             → (Ǝ i ꞉ I , small-↓ᴮ (γ i) ↠ ↓ᴮ L β a) holds
+        demote-equiv-to-surj =
+         ∥∥-functor (λ (i , f , f-is-equiv)
+                     → (i , f , equivs-are-surjections f-is-equiv))
+
+    H : (a : ⟨ L ⟩) → Γ ϕ (bounded-implies-local ϕ bnd) a ＝ f a
+    H a = reindexing-along-equiv-＝-sup
+           L ↓ᴮ-fa-equiv-↓ (β ∘ (↓-to-base ϕ a))
+           (Γ ϕ (bounded-implies-local ϕ bnd) a) (f a)
+           (sup-of-small-fam-is-lub L (β ∘ ↓-to-base ϕ a)
+                                    (bounded-implies-local ϕ bnd a))
+           (is-supᴮ (f a))
+     where
+      ↓ᴮ-fa-equiv-↓ : (small-↓ᴮ (f a)) ≃ (ϕ ↓ a)
+      ↓ᴮ-fa-equiv-↓ = Σ-cong ↓ᴮ-fa-equiv-S'
+       where
+        ↓ᴮ-fa-equiv-S' : (b : B)
```

Comment:
Since `S` is just some private variable above, I would give `↓ᴮ-fa-equiv-S'` and its cousins another name, maybe just `↓ᴮ-fa-equiv`.